project/src/test/java/com/google/sps/servlets/QueryServletTest.java

```diff
@@ -164,23 +173,62 @@ public void getRequest_userPreferencesForwardedToFetcher() throws Exception {
         .setCuisines(ImmutableList.of("sushi", "hamburger"))
         .build();
 
-    servlet.doGet(REQUEST, RESPONSE);
+    servlet.doPost(REQUEST, RESPONSE);
 
     verify(FETCHER).fetch(expectedUserPrefs);
   }
 
   @Test
   // This test checks that the PlacesScorer is called with the expected parameters
-  public void getRequest_placesAndUserLocationForwadedToScorer() throws Exception {
+  public void postRequest_placesAndUserLocationForwadedToScorer() throws Exception {
     ImmutableList<Place> places = createPlacesListBySize(1);
     when(FETCHER.fetch(any(UserPreferences.class))).thenReturn(places);
     when(REQUEST.getParameter("location")).thenReturn("00.00000000,00.00000000");
 
-    servlet.doGet(REQUEST, RESPONSE);
+    servlet.doPost(REQUEST, RESPONSE);
 
     verify(SCORER).getScores(places, new LatLng(00, 00));
   }
 
+  @Test
+  // This test checks that storeUserPreferences is called with the expected parameters
+  // when the servlet gets a valid ID token
+  public void postRequest_validIdToken_userIdAndPreferencesForwardedForStoring() throws Exception {
+    when(REQUEST.getParameter("rating")).thenReturn("4");
```

Comment:
You already have initializeRequestParameters called in the before method, why reset this here?

If it's for clarity as for when you're building the UserPreferences object (which I get), you can have all of these variants as constants & have a method that returns the matching UserPreferences object.

Reply:
Yes it was for clarity :)
I changed it as you suggested